python/packages/zarathustra/skills/subgraph_query_abci/behaviours.py

```diff
@@ -175,21 +217,28 @@ def async_act(self) -> Generator:
 
         with self.context.benchmark_tool.measure(self.behaviour_id).local():
             sender = self.context.agent_address
-            config = self._check_config(self.params.config)
-            subgraph_url = self.params.config["subgraph_url"]
-            subgraph_query = self.params.config["subgraph_query"]
-            content = json.dumps({"subgraph_url": subgraph_url, "subgraph_query": subgraph_query})
-            self.context.logger.info(f"subgraph config: {content}")
+            subgraph_queries = self._verify_subgraph_config(self.params.config)
+            self.context.logger.info(f"subgraph queries: {subgraph_queries}")
+            content = serialize({"subgraph_queries": subgraph_queries})
             payload = LoadSubgraphComponentsPayload(sender=sender, content=content)
 
         with self.context.benchmark_tool.measure(self.behaviour_id).consensus():
             yield from self.send_a2a_transaction(payload)
             yield from self.wait_until_round_end()
 
-    def _check_config(self, config):
-        required = ("subgraph_api_key", "subgraph_url", "subgraph_query")
-        if not all(map(config.get, required)):
-            raise ValueError(f"Ensure all required parameters are provided: {required}")
+    def _verify_subgraph_config(self, config: dict) -> dict[str, SubgraphQueryConfig]:
+        if not config.get("subgraph_api_key"):
+            self.context.logger.warning("No subgraph_api_key provided!")
```

Comment:
you will still need to set this. Set to empty string. If your URL does not contain the `{api_key}` part in the string, the `str.format(api_key="")` will just leave the string unaltered as is